app/assets/javascripts/initializers/initScrolling.js

```diff
@@ -394,7 +394,9 @@ function fetchNextPageIfNearBottom() {
       fetchNextFollowingPage(indexContainer);
     };
   } else {
-    scrollableElemId = 'articles-list';
+    const isHome = indexWhich === 'home';
```

Comment:
Refactor (blocking): No need to introduce a `data-which` attribute. We can handle this by checking if `document.getElementById('main-content')` exists or not and set the scrollable element accordingly.